config/confighttp/confighttp.go

```diff
@@ -85,6 +111,16 @@ func (hcs *HTTPClientSettings) ToClient(ext map[config.ComponentID]component.Ext
 		}
 	}
 
+	// Compress the body using specified compression methods if non-empty string is provided.
+	// Supporting Gzip, zlib, deflate, snappy, and zstd.
+	if hcs.Compression != "" {
+		if compressionKey := getHTTPCompressionKey(hcs.Compression); compressionKey != CompressionUnsupported {
+			clientTransport = middleware.NewCompressRoundTripper(clientTransport, string(compressionKey))
+		} else {
+			return nil, fmt.Errorf("unsupported compression type %q", hcs.Compression)
```

Comment:
Should this return an error, or simply not use compression?  Both outcomes could be surprising and I'm not sure which is the right choice.

Reply:
The choice of compression is an explicit one, and not doing what the user is explicitly requesting would be an error, IMO.

Reply:
I agree with @jpkrohling